### ISSUE TITLE
secret key to manage influxdb locally

### ISSUE THREAD

**Issue by n1tr0g** [influxdata/influxdb#2278]
Hey,

inspired by varnish design we could have a file containing the administration secret. Basically varnish is using this mechanism for local administration tasks only. One major benefit of that is that you can manage/view this file with external tools such as chef, puppet, ansible, etc. So if you know the content of the secret file and you are accessing the server locally (127.0.0.1) you can run administration tasks.

To make this work we would probably need to add another header/parameter to authenticate based on the secret file. 

To make it more secure we could put some restrictions in place:
- file has to be rw only by the user running influxdb
- secret file authentication is only available via localhost

Without that functionality automatically managing the root user password itself is really hard,  because you need to keep the current and future password to make it work.

That's the alternative to https://github.com/influxdb/influxdb/issues/2193

Probably it's a bit confusing so let me know if you have more question.

**Comment by beckettsean** [comment]
This feels like our preferred solution to the empty user problem at first authentication, rather than #2193.

**Comment by corylanou** [comment]
As long as this doesn't create external dependency issues, or cross platform issues, I'm fine with it.

**Comment by neonstalwart** [comment]
what about this idea:

```toml
[http]

  [http.authenticated]
  bind-address = "0.0.0.0"
  port = 8086
  authenticated = true
  ssl-cert = "/path/to/cert.pem"

  [http.open]
  bind-address = "127.0.0.1"
  port = 8087
  authenticated = false
```

the idea is that you can have multiple http servers running.  the names `http.open` and `http.authenticated` are not mandated - they are really just for self-documenting the config.  you can have more than 2 http bindings if you want (http vs https or whatever).  in this example the authenticated server is bound to 0.0.0.0 to make it public (usually you might probably bind to 127.0.0.1 and have a reverse proxy instead) and the unauthenticated server is only bound to 127.0.0.1 and using http://localhost:8087 you can create users that can then connect to https://server.com:8086.

the basis of this solution is simply to be able to configure multiple http bindings independently and the thing i like about this idea is that it provides a more general way to solve a number of problems - e.g. http and https bindings, etc.

**Comment by otoolep** [comment]
Yes, that might be a better way to do this, thanks @neonstalwart

**Comment by n1tr0g** [comment]
so if that's the case are we closing https://github.com/influxdb/influxdb/pull/2530 ? 

I like both ideas as long one of them gem implemented and released with 0.9.* because without a feature like that doing config management for influxdb is really painful.

**Comment by otoolep** [comment]
No decisions made yet, we'll definitely keep this ticket updated @n1tr0g

**Comment by neonstalwart** [comment]
to add more detail to my thought... i was thinking that the admin server config would get rolled into this too.

```toml
[http]

  [http.authenticated]
  # type = "api"
  # hostname = ""
  bind-address = "0.0.0.0"
  port = 8086
  authenticated = true
  ssl-cert = "/path/to/cert.pem"

  [http.open]
  # type = "api"
  bind-address = "127.0.0.1"
  port = 8087
  authenticated = false

  [http.admin]
  type = "admin"
  bind-address = "127.0.0.1"
  port = 8083
```

might want to also enable control of logging (to separate files?) per server too.

**Comment by dmke** [comment]
What is the current status of this?

**Comment by dgnorton** [comment]
closing: implemented JWT token authentication in #6609
